src-extras/Database/LSMTree/Extras/NoThunks.hs

```diff
@@ -506,7 +506,7 @@ instance NoThunks a => NoThunks (MutVar s a) where
   showTypeOf (_ :: Proxy (MutVar s a)) = "MutVar"
   wNoThunks ctx var = do
       x <- unsafeSTToIO $ readMutVar var
-      noThunks ctx (AllowThunk x) -- TODO: atomicModifyMutVar' is not strict enough
+      noThunks ctx x
```

Comment:
So what happened with the problem of `atomicModifyMutVar'`?

Reply:
We used to use it for `RefCounter`, but now we use a `PrimVar` instead, so we're not checking thunks in `MutVar`s anymore

Reply:
The problem is still there -- but not allowing a thunk is the proper instance anyway